files/scripts/cluster_provision.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+if [ "$#" -gt 4 ] || [ "$#" -lt 3 ]; then
+	echo "Illegal number of arguments !"
+	echo "cluster_provision.sh [WORKSPACE] [machine_type] [scripts_branch] -v" 
+	echo "[required] optional"
+	exit 1
+fi
+
+if [ "${4}" == '-v' ]; then
+	set -ex
+fi
+
+WORKSPACE=$1
+machine_type=$2
+scripts_branch=$3
+
+if [ ${machine_type} == "qdc" ]; then
+	machine_list="qdcohpc, qdc01, qdc02, qdc03"
+	preseed_type="kickstart"
+	kernel_desc="CentOS 7.5"
+	initrd_desc="CentOS 7.5"
+	preseed_name="CentOS Upstream"
+	kernel_opts="ip=dhcp text inst.stage2=http://mirror.centos.org/altarch/7/os/aarch64/ inst.repo=http://mirror.centos.org/altarch/7/os/aarch64/ inst.ks=file:/ks.cfg"
+elif [ ${machine_type} == "d05" ]; then
+	machine_list="d05ohpc, d0301, d0302, d0303"
+	preseed_type="kickstart"
+	kernel_desc="CentOS ERP 18.06"
+	initrd_desc="CentOS ERP 18.06"
+	preseed_name="CentOS"
+	kernel_opts="ip=dhcp text inst.stage2=http://releases.linaro.org/reference-platform/enterprise/18.06/centos-installer/ inst.repo=http://mirror.centos.org/altarch/7/os/aarch64/ inst.ks=file:/ks.cfg"
+fi
+
+# Chose known good kernel/initrd/cmdline
+
+cd ${WORKSPACE}
```

Comment:
do not change dir, cat > $WORKSPACE/machine_provision

cd doesn't work well with errors, and pushd/popd are a poor replacement for using the full path every time